Source/gg2/Objects/InGameElements/Character.events/Step.xml

```diff
@@ -76,6 +78,14 @@ else
     y -= vspeed;
 }
 
+// Fallback
```

Comment:
Having a second ground test is why it's fallback. Wouldn't be a problem if GM didn't do automatic motion. (Note that we need to use hspeed/vspeed and not our own variables, in order for GM's automatic "get outside of solid" to work, if it's still depended on anywhere.) In fact, I think undoing GM's automatic motion should be done at the very end of the block, but that separates it from our manual motion.